impl/src/main/java/org/jboss/weld/bean/builtin/InstanceImpl.java

```diff
@@ -151,9 +151,15 @@ public boolean isUnsatisfied() {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public <X> WeldInstance<X> select(Type subtype, Annotation... qualifiers) {
-        // TODO Auto-generated method stub
-        return null;
+        // verify if this was invoked on WeldInstance<Object>
+        if (!this.getType().getTypeName().equals(Object.class.getName())) {
```

Comment:
`!this.getType().equals(Object.class)` should be fine...

Reply:
Hmm yep, it eluded me yet again that Class is in fact Type :)